xchange-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingService.java

```diff
@@ -0,0 +1,147 @@
+                                break;
+                            case unsubscribed:
+                                LOG.info("Channel {} has been unsubscribed", channelName);
+                                channels.remove(statusMessage.getChannelID());
+                                break;
+                            case error:
+                                LOG.error("Channel {} has been failed: {}", channelName, statusMessage.getErrorMessage());
+                        }
+                        break;
+                    default:
+                        LOG.warn("Unexpected event type has been received: {}", krakenEvent);
+                }
+                return;
+
+            }
+        } catch (JsonProcessingException e) {
+            LOG.error("Error reading message: {}", e.getMessage(), e);
+        }
+
+        if (!message.isArray() || channelName == null) {
+            LOG.error("Unknown message: {}", message.asText());
+            return;
+        }
+
+        super.handleMessage(message);
+    }
+
+    @Override
+    protected String getChannelNameFromMessage(JsonNode message) throws IOException {
+        String channelName = null;
+        if (message.has("channelID")) {
+            channelName = channels.get(message.get("channelID").asInt());
+        }
+
+        if (message.isArray() && message.get(0).isInt()) {
+            channelName = channels.get(message.get(0).asInt());
+        }
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("ChannelName {}", StringUtils.isBlank(channelName) ? "not defined" : channelName);
+        }
+        return channelName;
+    }
+
+
+    @Override
+    public String getSubscribeMessage(String channelName, Object... args) throws IOException {
+        String [] channelData = channelName.split(KrakenStreamingMarketDataService.KRAKEN_CHANNEL_DELIMITER);
+        String pair = channelData[1];
+        KrakenSubscriptionName subscriptionName = KrakenSubscriptionName.valueOf(channelData[0]);
+
+        Integer depth = null;
+        if (args.length > 0 && args[0] != null) {
+            depth = (Integer) args[0];
+        }
+        int reqID = Math.abs(UUID.randomUUID().hashCode());
+        subscriptionRequestMap.put(reqID, channelName);
+
+        KrakenSubscriptionMessage subscriptionMessage = new KrakenSubscriptionMessage(reqID, subscribe,
+                Collections.singletonList(pair), new KrakenSubscriptionConfig(subscriptionName, depth));
+        return objectMapper.writeValueAsString(subscriptionMessage);
+    }
+
+    @Override
+    public String getUnsubscribeMessage(String channelName) throws IOException {
+        String [] channelData = channelName.split(KrakenStreamingMarketDataService.KRAKEN_CHANNEL_DELIMITER);
+        String pair = channelData[1];
+        KrakenSubscriptionName subscriptionName = KrakenSubscriptionName.valueOf(channelData[0]);
+
+        int reqID = Math.abs(UUID.randomUUID().hashCode());
```

Comment:
This made me smile. No change needed; I just love this as a way of making a random number and felt like sharing.

Reply:
:)